src/graph.rs

```diff
@@ -111,56 +105,60 @@ impl<'tree> Graph<'tree> {
                 let ctx = self.1;
                 let mut seq = serializer.serialize_seq(Some(graph.graph_nodes.len()))?;
                 for (node_index, node) in graph.graph_nodes.iter().enumerate() {
-                    seq.serialize_element(&JSONNode(node_index, node, ctx))?;
+                    seq.serialize_element(&InContext(&(node_index, node), ctx))?;
                 }
                 seq.end()
             }
         }
 
-        impl<'a> ser::Serialize for JSONNode<'a> {
+        impl<'a> ser::Serialize for InContext<'a, (usize, &'a GraphNode)> {
             fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
             where
                 S: serde::Serializer,
             {
-                let node_index = self.0;
-                let node = self.1;
-                let ctx = self.2;
+                let node_index = self.0 .0;
+                let node = self.0 .1;
```

Comment:
Are the spaces before the dots intentional here?

Reply:
I think so, as otherwise the lexer would try to recognize it as a float. You can also do `(self.0).1` but it works without the parens if you add a space.

Reply:
Oh, wow

Reply:
The autoformatter inserts these spaces on my behalf, FWIW.

Reply:
If the autoformatter says this is the way to go, this is the way to go :smile: